apps/web/pages/event-types/[type]/index.tsx

```diff
@@ -473,6 +473,19 @@ const EventTypePage = (props: EventTypeSetupProps) => {
       }
     }
 
+    // Prevent two payment apps to be enabled
+    let enabledPaymentApps = 0;
+    for (const appKey in metadata?.apps) {
+      const app = metadata?.apps[appKey];
+      if (app.price && app.enabled) {
+        enabledPaymentApps++;
+      }
+
+      if (enabledPaymentApps > 1) {
+        throw new Error(t("event_setup_multiple_payment_apps_error"));
+      }
+    }
```

Comment:
`suggestion`: Something like this could be cleaner.

```suggestion
    const enabledPaymentApps = Object.values(metadata?.apps || {}).filter(app => app.price && app.enabled).length;
if (enabledPaymentApps > 1) {
  throw new Error(t("event_setup_multiple_payment_apps_error"));
}
```

Reply:
typescript issue 
<img width="1946" alt="Screenshot 2023-11-08 at 10 46 48" src="https://github.com/calcom/cal.com/assets/33722304/5085963b-c4dd-4334-97f7-a313aa73b315">

Reply:
yes it would indeed iterate over all the apps inside the metadata, both approach are fine on such small dataset, both work just a matter of preference if you like the shorter version or not up to you.

I would extract this logic in a small function in the same file with clear naming and just the required parameters 

preventMultiplePaymentApps(...)

also there is a type error you can fix the same way you did in the other place ` const app = metadata?.apps[appKey as keyof AppMetadata];`